R/find_bad_apples.R

```diff
@@ -2,23 +2,98 @@
-#' 'Indices' (list of row indices with outliers),
-#' 'Total Outliers' (number of outliers in the column)
+#' @return A dataframe with columns for 'variable' (dataframe column name),
+#' 'total_outliers' (number of outliers in the column), and
+#' 'indices' (list of row indices with outliers)
 #'
 #' @examples
 #' find_bad_apples(df)
 #'
 #' @export
 
 find_bad_apples <- function(df) {
-  # insert code here
+
+    # Returns a warning and stops the function if the dataframe includes non-numeric values
+    nums <- unlist(lapply(df, is.numeric))
+    if (dim(df)[2] != sum(nums)){
+        warning('Every column in your dataframe must be numeric.')
+        break
+
+        # Returns a warning and stops the function if the dataframe has fewer than 30 rows
+    } else {
+        if (dim(df)[1] < 30){
+            warning('Sorry, you don\'t have enough data. The dataframe needs to have at least 30 rows.')
+            break
+
+        } else {
+            # Creates empty output dataframe that will be added to
+            output <- tibble(variable = character(),
+                             indices = tibble(),
+                             total_outliers = integer()) %>%
+                group_by(variable, total_outliers) %>%
+                nest()
+
+            columns = names(df)
+            c <- 1
+
+            for (column in columns){
+
+                # Finds the mean and standard deviation for each column in the dataframe
+                mean = mean(df[,c])
+                sd = sd(df[,c])
+
+                col <- 1
+                ind <- c()
+                tot <- 0
+
+                r <- 1
+
+                # Evaluates each value in a column to see if it's an outlier
+                values = df[,c]
+                for(value in values){
+                    if (((mean - 2*sd) <= value & value <= (mean + 2*sd)) == TRUE){
+                        r <- r + 1
+                    } else {
+
+                        # Adds the indices of any found outliers to 'ind'
+                        # And adds 1 to the total number of outliers for each outlier found
+                        if (((mean - 2*sd) <= value & value <= (mean + 2*sd)) == FALSE){
+                            ind <- c(ind, r)
+                            tot <- tot + 1
+                            r <- r + 1
+                        } else {
+                            next
+                        }
+                    }
+                }
+                c <- c + 1
+                col <- col + 1
+
+                if (tot == 0){
+                    next}
+                else {
+                    # Creates a row for each column with outliers
+                    addition <- tibble(variable = column,
+                                       indices = ind,
+                                       total_outliers = tot) %>%
+                        group_by(variable, total_outliers) %>%
+                        nest()
```

Comment:
specify where functions came from for non base R.

Reply:
Will do!